src/redisearch_rs/buffer/src/lib.rs

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! The `Buffer` API Rust wrappers.
+//!
+//! This crate provides Rust wrappers for the `Buffer`, `BufferReader` and `BufferWriter` structs
+//! from `buffer.h`.
+
+use std::{
+    cmp,
+    ptr::{NonNull, copy_nonoverlapping},
+    slice,
+};
+
+/// Redefines the `Buffer` struct from `buffer.h`
+#[repr(C)]
+pub struct Buffer {
```

Comment:
As we discussed last week, let's make this a:

```rust
#[repr(transparent)]
pub struct Buffer(ffi::Buffer);
```

using bindgen.

Reply:
Same strategy for the other two structs.

Reply:
yeah, was the plan but then MIRI came in the way. :cry: